nucleus/admin/util/src/main/java/com/sun/enterprise/admin/remote/RemoteAdminCommand.java

```diff
@@ -978,27 +979,24 @@ public String getUsage() {
      * @return the URI so far, including the newly-added option
      */
     private StringBuilder addStringOption(StringBuilder uriString, String name, String option) {
-        try {
-            String encodedOption = URLEncoder.encode(option, "UTF-8");
-            uriString.append(name).append('=').append(encodedOption).append(QUERY_STRING_SEPARATOR);
-        } catch (UnsupportedEncodingException e) {
-            // XXX - should never happen
-            throw new RuntimeException("Error encoding value for: " + name + ", value:" + option, e);
-        }
+        String encodedOption = URLEncoder.encode(option, UTF_8);
+        uriString.append(name).append('=').append(encodedOption).append(QUERY_STRING_SEPARATOR);
         return uriString;
     }
 
     /**
      * Add a password option, passing it as a header in the request
+     *
+     * @deprecated insecure!!!
      */
+    @Deprecated
     private StringBuilder addPasswordOption(StringBuilder uriString, String name, String option) throws IOException {
         if (passwordOptions == null) {
             passwordOptions = new StringBuilder();
         } else {
             passwordOptions.append(QUERY_STRING_SEPARATOR);
         }
-        GFBase64Encoder encoder = new GFBase64Encoder();
-        passwordOptions.append(name).append('=').append(URLEncoder.encode(encoder.encode(option.getBytes()), "UTF-8"));
+        passwordOptions.append(name).append('=').append(Base64.getUrlEncoder().encodeToString(option.getBytes(UTF_8)));
```

Comment:
Why is the `base64url` encoding left here?

Reply:
Here it should be ok, because it is used as a part of URL